pwstore-fast is already in Debian and is reportedly much faster.

Index: happstack-authenticate-0.10.15/happstack-authenticate.cabal
===================================================================
--- happstack-authenticate-0.10.15.orig/happstack-authenticate.cabal	2014-12-23 13:59:58.912423502 +0100
+++ happstack-authenticate-0.10.15/happstack-authenticate.cabal	2014-12-23 13:59:58.912423502 +0100
@@ -44,7 +44,7 @@
                        fb                           >= 0.13 && < 1.1,
                        safecopy                     >= 0.6,
                        mtl                          >= 2.0,
-                       pwstore-purehaskell          == 2.1.*,
+                       pwstore-fast,
                        QuickCheck                   >= 2,
                        text                         >= 0.11 && < 1.3,
                        time                         >= 1.2 && < 1.6,
